src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1044,18 +1044,18 @@ private void OnAsyncTaskStateChanged(DynamoScheduler sender, TaskStateChangedEve
 
                         if (Logging.Analytics.ReportingAnalytics)
                         {
-                            var modifiedNodes = "";
                             if (updateTask.ModifiedNodes != null && updateTask.ModifiedNodes.Any())
                             {
-                                modifiedNodes = updateTask.ModifiedNodes
-                                    .Select(n => n.GetOriginalName())
-                                    .Aggregate((x, y) => string.Format("{0}, {1}", x, y));
+                                // Send analytics for each of modified nodes so they are counted individually
+                                foreach (var node in updateTask.ModifiedNodes)
+                                {
+                                    // Notice the executionTimeSpan is still for the whole delta computation
+                                    Dynamo.Logging.Analytics.TrackTimedEvent(
+                                        Categories.Performance,
+                                        e.Task.GetType().Name,
+                                        executionTimeSpan, node.GetOriginalName());
+                                }
```

Comment:
@mmisol @aparajit-pratap Thank you for your input. I would like to take more time to think through how we track real changes to nodes executed.

From what I can tell, adding/deleting/moving/modifying a node will all trigger execution so even a node showing as popularly executed, it might be just users modify them regularly.. Like Code Block node

Reply:
Certain cases, undo/redo may trigger multiple runs..

Reply:
Also @mmisol I dont think I can only look for ModifiedNodes.count = 1 case because when upstream node modified, we usually make downstream nodes dirty as well in Dynamo. e.g. with the following example, if user touches CBN, the all three nodes will be in ModifiedNodes set during execution.. Maybe the `ModifiedNodes` name itself is misleading..
![image](https://user-images.githubusercontent.com/3942418/101865993-bf1c0980-3b45-11eb-86ae-7e7d9ac3ed56.png)

Reply:
maybe a stupid question... doesn't this send `n` http requests for each graph run? Doesn't that seem potentially slow with large graphs?

Reply:
@mjkkirschner ADPSDK will handle the requests with optimization on their end. From my dottrace profiling, this do not seem impact Dynamo performance that much. I have a screen shot of that in PR description using DotTrace.